### ISSUE TITLE
Inconsistent wrapping of body of implicit-return arrow functions

### ISSUE THREAD

**Issue by kylehg** [prettier/prettier#2446]
If I use an arrow function that returns a single expression, it wraps as I'd expect, with the params on the initial line and the body indented (similar to how non-implicit-return arrow functions wrap. (Both examples show the post-processed code—see sandbox link below.)

```js
// This wraps as I'd expect
array.filter((element, index) =>
  element.field.some(someVeryLongPredicateFunction)
);
```

But if the function body is some kind of operator expression, it wraps the params as well

```js
// This wraps strangely
array.filter(
  (el, index) => somePredicate(index) && el.field.some(someLongPredicate)
);
```

This seems strangely inconsistent. Is this intentional?

[See sandbox](https://prettier.io/playground/#%7B%22content%22%3A%22%2F%2F%20This%20wraps%20as%20I'd%20expect%20%2F%20write%20myself%5Cnarray.filter((element%2C%20index)%20%3D%3E%5Cn%20%20element.field.some(someVeryLongPredicateFunction)%5Cn)%5Cn%5Cn%2F%2F%20This%20wraps%20strangely%5Cnarray.filter((element%2C%20index)%20%3D%3E%20%5Cn%20%20somePredicate(index)%20%26%26%20element.field.some(someVeryLongPredicate)%5Cn)%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D)

**Comment by vjeux** [comment]
This is not intended. This is likely the same issue as https://github.com/prettier/prettier/issues/2456

**Comment by azz** [comment]
It's not _exactly_ the same issue as #2456 (not fixed in #2486), but they are quite similar.

To fix this we'd need to allow logical expressions containing a function/arrow expression to hug.

**Comment by thorn0** [comment]
Closing in favor of #3465
